### PR TITLE
Remove vagrant support for OS packaging tests

### PR DESCRIPTION
**Pull request by breskeby**
The vagrant related build logic hasn't been maintained in a while and the vagrant gradle plugin also broke compatibiliy
for gradle --configuration-cache.
To remove the surface of maintenance we remove support for running packaging tests via vagrant

### CODE REVIEW

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/DistroTestPlugin.java [hunk 1]
```diff
@@ -164,82 +147,15 @@ public void apply(Project project) {
...
-                    }, upgradeDepsTask);
+                    }, distribution, bwcDistro);
                     versionTasks.get(version.toString()).configure(t -> t.dependsOn(upgradeTest));
-                    upgradeTestTasks.computeIfAbsent(version.toString(), k -> new ArrayList<>()).add(upgradeTest);
                 }
             }
         }
-
-        // setup jdks used by no-jdk tests, and by gradle executing
-        TaskProvider<Copy> linuxGradleJdk = createJdk(project, "gradle", GRADLE_JDK_VENDOR, GRADLE_JDK_VERSION, "linux", "x64");
```
Comment (mark-vieira) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/DistroTestPlugin.java:182]: I think we can ditch these constants as well.

**Review by mark-vieira — APPROVED** [review]
LGTM. I'm very happy to see this go.

**Comment by mark-vieira** [comment]
cc @rjernst because we bring up the topic of how to reproduce these tests all the time. I think using [this tool](https://github.com/elastic/elasticsearch-infra/tree/master/buildkite-tools) is the answer to that question. @breskeby we should probably update the testing doc to mention this. It's internal-only, but I think that's good enough.

**Comment by rjernst** [comment]
If we have a well defined way to reproduce failures in buildkite now, I'll be ecstatic for the vagrant based runner to be removed. 🥳

**Comment by breskeby** [comment]
I've tested using the buildkite tools that I linked now in the testing doc and it is so much better and more convenient than how it was before to debug ci issues in general and especially OS specific packaging issues.

**Comment by breskeby** [comment]
This work was partially related to https://github.com/elastic/elasticsearch/issues/57918

**Pull request merged**